File: 11-lecture/live/TestBlock.cpp
#include <iostream>
using namespace std;

// Test block scope and scope shadowing.
// Test for loop block scope.


int main()
{

    int num = 100;

    // Block scope
    {

        int num = 50;
        cout << num << endl;
        int x = 42;
        cout << x << endl;

    }

    cout << num << endl;




}

File: 11-lecture/live/TestBlock.java
public class TestBlock
{
    public static void main(String[] args)
    {
        int num = 50;

        {
            int num = 100;

            System.out.println("In block, num is: " + num);
        }

        for (int i = 10; i > 0 ; i--)
        {
            int num = 400;
        }


        System.out.println("After block, num is: " + num);
    }
}